mern/client/src/screens/RecoverSession/recoverSession.js

```diff
@@ -1,105 +1,123 @@
+  const [, setNavState] = useContext(NavStateContext);
 
-  handleChange(event) {
-    this.setState({ value: event.target.value });
+  const PageChange = (url) => {
+  Navigate(url);
   }
-  handleSubmit(event) {
-    alert("A name was submitted: " + this.state.value);
+
+  const handleSubmit = (event) => {
     event.preventDefault();
+    // show navbar buttons for starting-intent page
+    setNavState(true) 
+    alert(`The session ID you entered was: ${sessionid}`)
   }
-  handleBack(event) {
-    this.props.navigate("/");
+
+  const handleChange = (event) => {
+    // set sessionid to input text
+    setSessionID(event.target.value);
   }
 
-  render() {
     return (
       <div className="container">
         <h1 className="pageTitle">Recover Session</h1>
-        <form className="inputForm w-1/3 mx-auto" onSubmit={this.handleSubmit}>
+        <form className="inputForm w-1/3 mx-auto" onSubmit={handleSubmit}>
           <label className="text-xl text-white font-nunito font-medium">
-            {" "}
-            Input Session ID{" "}
+            Input Session ID
           </label>
           <input
             className="m-6 w-3/4 pl-4 pr-2 pt-2 pb-2 block rounded-full text-black"
             type="text"
-            value={this.state.value}
-            onChange={this.handleChange}
+            value={sessionid}
+            onChange={handleChange}
             placeholder="Session ID"
           />
           <GenericButton
             buttonType="white"
-            onClick={() => this.handleChange}
+            onClick={() => handleSubmit}
             disabled={false}
             text={"Begin Session"}
           />
         </form>
         <div className="buttonContainer">
           <GenericButton
             buttonType="outline"
-            onClick={this.handleBack}
+            onClick={() => {
+              PageChange("/");
+              setNavState(false);
+            }}
             disabled={false}
             text={"Go Back"}
           />
         </div>
       </div>
     );
   }
-}
 
-export default withRouter(RecoverSession);
+// CLASS COMPONENT VRSN:
+// class RecoverSession extends Component {
 
-// FUNCTIONAL REACT VRSN:
-// function RecoverSession() {
-//   const Navigate = useNavigate();
-//   const PageChange = (url) => {
-//   Navigate(url);
-//   }
+//   constructor(props) {
+//     super(props);
 
-//     const [name, setName] = useState(0);
+//     this.state = { value: "" };
+//     this.handleChange = this.handleChange.bind(this);
+//     this.handleSubmit = this.handleSubmit.bind(this);
+//     this.handleBack = this.handleBack.bind(this);
+//   }
 
-//     const handleSubmit = (event) => {
-//       event.preventDefault();
-//       alert(`The name you entered was: ${name}`)
-//       console.log(name)
-//     }
+//   handleChange(event) {
+//     this.setState({ value: event.target.value });
+//   }
+//   handleSubmit(event) {
+//     alert("A name was submitted: " + this.state.value);
```

Comment:
Let's delete the class components version. Unlikely we will need it and the comments take up dead space.